src/controllers/subscriptions.js

```diff
@@ -80,17 +61,20 @@ exports.getResource = function * fetch () {
   id = id.toLowerCase()
   log.debug('fetching subscription ID ' + id)
 
+  const requestOwner = uri.make('account', this.req.user.name)
   const subscription = yield Subscription.findById(id)
-  if (subscription) {
-    this.body = subscription.getDataExternal()
-  } else {
+  if (!subscription) {
     throw new NotFoundError('Unknown subscription ID')
+  } else if (!(requestOwner === subscription.owner || this.req.user.is_admin)) {
```

Comment:
This pattern repeats a few times, maybe we should make a function `isAuthorized(this.req.user, subscription.owner/subscription.subject)`